pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -464,6 +464,12 @@ func (s *DiscoveryServer) ready(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 	}
+	if s.KubeController != nil {
+		if !s.KubeController.HasSynced() {
+			w.WriteHeader(503)
+			return
+		}
+	}
```

Comment:
We have rediness probe for pilot, it's not until kube service registry and config registry has synced, pilot can serve request.